src/core/lib/channel/channel_stack_builder.cc

```diff
@@ -25,252 +25,59 @@
-  b->begin.filter = nullptr;
-  b->end.filter = nullptr;
-  b->begin.next = &b->end;
-  b->begin.prev = &b->end;
-  b->end.next = &b->begin;
-  b->end.prev = &b->begin;
-  b->name = name;
-  return b;
-}
-
-void grpc_channel_stack_builder_set_target(grpc_channel_stack_builder* b,
-                                           const char* target) {
-  gpr_free(b->target);
-  b->target = gpr_strdup(target);
-}
-
-std::string grpc_channel_stack_builder_get_target(
-    grpc_channel_stack_builder* b) {
-  return b->target == nullptr ? std::string("unknown") : std::string(b->target);
-}
-
-static grpc_channel_stack_builder_iterator* create_iterator_at_filter_node(
-    grpc_channel_stack_builder* builder, filter_node* node) {
-  grpc_channel_stack_builder_iterator* it =
-      static_cast<grpc_channel_stack_builder_iterator*>(
-          gpr_malloc(sizeof(*it)));
-  it->builder = builder;
-  it->node = node;
-  return it;
-}
-
-void grpc_channel_stack_builder_iterator_destroy(
-    grpc_channel_stack_builder_iterator* it) {
-  gpr_free(it);
-}
-
-grpc_channel_stack_builder_iterator*
-grpc_channel_stack_builder_create_iterator_at_first(
-    grpc_channel_stack_builder* builder) {
-  return create_iterator_at_filter_node(builder, &builder->begin);
-}
-
-grpc_channel_stack_builder_iterator*
-grpc_channel_stack_builder_create_iterator_at_last(
-    grpc_channel_stack_builder* builder) {
-  return create_iterator_at_filter_node(builder, &builder->end);
-}
-
-bool grpc_channel_stack_builder_iterator_is_end(
-    grpc_channel_stack_builder_iterator* iterator) {
-  return iterator->node == &iterator->builder->end;
-}
-
-const char* grpc_channel_stack_builder_iterator_filter_name(
-    grpc_channel_stack_builder_iterator* iterator) {
-  if (iterator->node->filter == nullptr) return nullptr;
-  return iterator->node->filter->name;
-}
-
-bool grpc_channel_stack_builder_move_next(
-    grpc_channel_stack_builder_iterator* iterator) {
-  if (iterator->node == &iterator->builder->end) return false;
-  iterator->node = iterator->node->next;
-  return true;
-}
-
-bool grpc_channel_stack_builder_move_prev(
-    grpc_channel_stack_builder_iterator* iterator) {
-  if (iterator->node == &iterator->builder->begin) return false;
-  iterator->node = iterator->node->prev;
-  return true;
-}
+namespace grpc_core {
 
-grpc_channel_stack_builder_iterator* grpc_channel_stack_builder_iterator_find(
-    grpc_channel_stack_builder* builder, const char* filter_name) {
-  GPR_ASSERT(filter_name != nullptr);
-  grpc_channel_stack_builder_iterator* it =
-      grpc_channel_stack_builder_create_iterator_at_first(builder);
-  while (grpc_channel_stack_builder_move_next(it)) {
-    if (grpc_channel_stack_builder_iterator_is_end(it)) break;
-    const char* filter_name_at_it =
-        grpc_channel_stack_builder_iterator_filter_name(it);
-    if (strcmp(filter_name, filter_name_at_it) == 0) break;
+ChannelStackBuilder::~ChannelStackBuilder() {
+  if (args_ != nullptr) {
```

Comment:
No need to check this here, because it's already checked inside of `grpc_channel_args_destroy()`.

Reply:
done